From c6d42b0f474539e5d8d27eda81655555756cafcb Mon Sep 17 00:00:00 2001
From: fanglifei <fanglifei@eswincomputing.com>
Date: Mon, 9 Sep 2024 10:17:35 +0800
Subject: [PATCH 257/652] WIN2030-16035: feat(eth): Adaption for ethernet d2d

Changelogs:
1. Fix dumpstack of d1 eth probe
2. Using the same delay value for d1 eth

Change-Id: I6059b2b5960d749551bb5d549f4721d9bd663f49
Signed-off-by: fanglifei <fanglifei@eswincomputing.com>
---
 arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi | 4 ++--
 drivers/net/ethernet/stmicro/stmmac/dwmac-win2030.c   | 7 ++++---
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
index ae2633ca6832a..fde17194951ad 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
@@ -1507,7 +1507,7 @@ d1_gmac0: ethernet@70400000 {
 			clock-names = "app", "stmmaceth","tx";
 			resets = <&d1_reset HSPDMA_RST_CTRL SW_HSP_ETH0_ARSTN>;
 			reset-names = "ethrst";
-			// iommus = <&smmu1 WIN2030_SID_ETH0>;
+			iommus = <&smmu1 WIN2030_SID_ETH0>;
 			tbus = <WIN2030_TBUID_ETH>;
 			dma-noncoherent;
 			eswin,hsp_sp_csr = <&d1_hsp_sp_csr 0x1030 0x100 0x108>;
@@ -1537,7 +1537,7 @@ d1_gmac1: ethernet@70410000 {
 			clock-names = "app", "stmmaceth","tx";
 			resets = <&d1_reset HSPDMA_RST_CTRL SW_HSP_ETH1_ARSTN>;
 			reset-names = "ethrst";
-			// iommus = <&smmu1 WIN2030_SID_ETH1>;
+			iommus = <&smmu1 WIN2030_SID_ETH1>;
 			tbus = <WIN2030_TBUID_ETH>;
 			dma-noncoherent;
 			eswin,hsp_sp_csr = <&d1_hsp_sp_csr 0x1034 0x200 0x208>;
diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac-win2030.c b/drivers/net/ethernet/stmicro/stmmac/dwmac-win2030.c
index 66e0cc12ce44b..e3b6c28394b6f 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-win2030.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-win2030.c
@@ -205,7 +205,7 @@ static void dwc_qos_fix_speed(void *priv, unsigned int speed, unsigned int mode)
 	case SPEED_1000:
 		rate = 125000000;
 
-		if (dwc_priv->dev_id == 0) {
+		if ((dwc_priv->dev_id & 0x1) == 0) {
 			regmap_write(dwc_priv->hsp_regmap, 0x118, 0x800c8023);
 			regmap_write(dwc_priv->hsp_regmap, 0x11c, 0x0c0c0c0c);
 			regmap_write(dwc_priv->hsp_regmap, 0x114, 0x23232323);
@@ -226,7 +226,7 @@ static void dwc_qos_fix_speed(void *priv, unsigned int speed, unsigned int mode)
 	case SPEED_100:
 		rate = 25000000;
 
-		if (dwc_priv->dev_id == 0) {
+		if ((dwc_priv->dev_id & 0x1) == 0) {
 			regmap_write(dwc_priv->hsp_regmap, 0x118, 0x803f8050);
 			regmap_write(dwc_priv->hsp_regmap, 0x11c, 0x3f3f3f3f);
 			regmap_write(dwc_priv->hsp_regmap, 0x114, 0x50505050);
@@ -247,7 +247,7 @@ static void dwc_qos_fix_speed(void *priv, unsigned int speed, unsigned int mode)
 	case SPEED_10:
 		rate = 2500000;
 
-		if (dwc_priv->dev_id == 0) {
+		if ((dwc_priv->dev_id & 0x1) == 0) {
 			regmap_write(dwc_priv->hsp_regmap, 0x118, 0x0);
 			regmap_write(dwc_priv->hsp_regmap, 0x11c, 0x0);
 			regmap_write(dwc_priv->hsp_regmap, 0x114, 0x0);
@@ -468,6 +468,7 @@ static int dwc_qos_probe(struct platform_device *pdev,
 	plat_dat->bsp_priv = dwc_priv;
 	plat_dat->phy_addr = PHY_ADDR;
 	plat_dat->clks_config = dwc_clks_config;
+	plat_dat->bus_id = dwc_priv->dev_id;
 
 	return 0;
 }
-- 
2.51.0

